packages/block-editor/src/components/inserter/hooks/use-insertion-point.js

```diff
@@ -74,57 +73,34 @@ function useInsertionPoint( {
 	const { getSelectedBlock } = useSelect( blockEditorStore );
 	const { destinationRootClientId, destinationIndex } = useSelect(
 		( select ) => {
-			const {
-				getSelectedBlockClientId,
-				getBlockRootClientId,
-				getBlockIndex,
-				getBlockOrder,
-				getInserterInsertionPoint,
-			} = unlock( select( blockEditorStore ) );
-			const selectedBlockClientId = getSelectedBlockClientId();
+			const { getNextInsertionPoint } = select( blockEditorStore );
 			let _destinationRootClientId = rootClientId;
 			let _destinationIndex;
-			const insertionPoint = getInserterInsertionPoint();
+			const insertionPoint = getNextInsertionPoint();
 
 			if ( insertionIndex !== undefined ) {
 				// Insert into a specific index.
 				_destinationIndex = insertionIndex;
-			} else if (
-				insertionPoint &&
-				insertionPoint.hasOwnProperty( 'insertionIndex' )
-			) {
+			} else if ( insertionPoint ) {
 				_destinationRootClientId = insertionPoint?.rootClientId
 					? insertionPoint.rootClientId
 					: rootClientId;
-				_destinationIndex = insertionPoint.insertionIndex;
-			} else if ( clientId ) {
-				// Insert after a specific client ID.
-				_destinationIndex = getBlockIndex( clientId );
-			} else if ( ! isAppender && selectedBlockClientId ) {
-				_destinationRootClientId = getBlockRootClientId(
-					selectedBlockClientId
-				);
-				_destinationIndex = getBlockIndex( selectedBlockClientId ) + 1;
-			} else {
-				// Insert at the end of the list.
-				_destinationIndex = getBlockOrder(
-					_destinationRootClientId
-				).length;
```

Comment:
I think I should leave all of this in and address it in a follow-up to refactor further.